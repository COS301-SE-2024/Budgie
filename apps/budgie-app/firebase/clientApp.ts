import { initializeApp } from 'firebase/app';
import { getAnalytics } from 'firebase/analytics';
import { connectFirestoreEmulator, getFirestore } from 'firebase/firestore';
import {
  connectAuthEmulator,
  getAuth,
  signInWithEmailAndPassword,
} from 'firebase/auth';

const firebaseConfig = {
  apiKey: "AIzaSyCjt6UnO1y9MAy1pcZJZzud-cbLak8zxSM",
  authDomain: "budgieapp-70251.firebaseapp.com",
  databaseURL: "https://budgieapp-70251-default-rtdb.firebaseio.com",
  projectId: "budgieapp-70251",
  storageBucket: "budgieapp-70251.appspot.com",
  messagingSenderId: "844132380486",
  appId: "1:844132380486:web:c8b0d1ef92593145923f47",
  measurementId: "G-E7WEEV82H5"
};

const app = initializeApp(firebaseConfig);

export const db = getFirestore(app);
// connectFirestoreEmulator(db, '127.0.0.1', 8080);
export const analytics = getAnalytics(app);
export const auth = getAuth(app);
