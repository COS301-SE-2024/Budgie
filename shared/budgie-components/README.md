# shared-budgie-components

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-budgie-components` to execute the unit tests via [Jest](https://jestjs.io).
