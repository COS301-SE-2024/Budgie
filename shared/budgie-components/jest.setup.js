import {
  mockFirebaseAuth,
  mockNextImage,
  mockFirebaseFunctions,
  mockNextRouter,
} from './src/mocks/globalMocks';

jest.mock('firebase/auth', () => mockFirebaseAuth);
jest.mock('next/image', () => mockNextImage.Image);
jest.mock('firebase/functions', () => mockFirebaseFunctions);
jest.mock('next/navigation', () => mockNextRouter);
