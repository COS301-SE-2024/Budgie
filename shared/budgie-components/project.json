{
  "name": "shared-budgie-components",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "shared/budgie-components/src",
  "projectType": "library",
  "tags": [],
  "// targets": "to see all targets run: nx show project shared-budgie-components --web",
  "targets": {}
}
