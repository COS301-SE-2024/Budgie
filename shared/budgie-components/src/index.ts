// Use this file to export React client components (e.g. those with 'use client' directive) or other non-server utilities
// export * from './lib/shared-budgie-components';

export { SignInModal } from './lib/sign-in-modal/SignInModal';
export { Navbar } from './lib/navbar/navbar';
export { NewNavbar } from './lib/new-navbar/new-navbar';
export { Settings } from './lib/settings-page/settings';
export { Dashboard } from './lib/dashboard/Dashboard';
export { Landing } from './lib/Landing/Landing';
export { SignUpPage } from './lib/sign-up-page/SignUpPage';
export { UserProvider, UserContext } from './lib/hooks/UserProvider';
export { OverviewPage } from './lib/overview-page/overview-page';
