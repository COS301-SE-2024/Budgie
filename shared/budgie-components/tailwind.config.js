const { createGlobPatternsForDependencies } = require('@nx/react/tailwind');
const { join } = require('path');

/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    join(
      __dirname,
      '{src,pages,components,app}/**/*!(*.stories|*.spec).{ts,tsx,html}'
    ),
    ...createGlobPatternsForDependencies(__dirname),
  ],
  theme: {
    extend: {
      fontFamily: {
        TripSans: 'TripSans',
      },
      colors: {
        BudgieGreen1: '#8EE5A2',
        BudgieGreen2: '#B4E5A2',
        BudgieGreen3: '#71C194',
        BudgieBlue: '#293652',
        BudgieWhite: '#F8F8F8',
        BudgieGray: '#E1E1E1',
        BudgieGrayLight: '#EAEAEA',
      },
    },
  },
  plugins: [],
};
