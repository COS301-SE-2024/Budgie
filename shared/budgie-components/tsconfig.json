{
  "compilerOptions": {
    "jsx": "react-jsx",
    "allowJs": false,
    "esModuleInterop": false,
    "allowSyntheticDefaultImports": true,
    "strict": true
  },
  "files": [],
  "include": [],
  "composite": true,
  "declaration": true,
  "references": [
    {
      "path": "./tsconfig.lib.json"
    },
    {
      "path": "./tsconfig.storybook.json"
    }
  ],
  "extends": "../../tsconfig.base.json"
}
