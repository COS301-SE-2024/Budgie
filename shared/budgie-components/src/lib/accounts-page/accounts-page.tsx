'use client';

import styles from './accounts-page.module.css';
import '../../root.css';
import { useEffect, useState } from 'react';
import { UploadStatementCSV } from '@capstone-repo/shared/budgie-components';
import { useRouter } from 'next/navigation';
/* eslint-disable-next-line */
export interface AccountsPageProps {}

//pages and modals------------------------------------------
interface NoAccountsPageProps {
  onAddClick: () => void;
}

function NoAccountsPage(props: NoAccountsPageProps) {
  return (
    <div className="mainPage">
      <div className="flex items-center justify-center bg-BudgieGray h-full w-full">
        <div
          onClick={props.onAddClick}
          className="flex flex-col items-center justify-center bg-BudgieGray hover:bg-black hover:bg-opacity-5 h-80 w-80 border-dashed border-2 border-gray-400 border-opacity-40 hover:border-opacity-100 cursor-pointer  rounded-3xl"
        >
          <span className="text-2xl text-gray-400 text-opacity-60">
            No Accounts Added Yet
          </span>
          <span className="mt-5 text-xl text-gray-400 text-opacity-60">
            Add an Account
          </span>
          <span
            className="mt-2 text-opacity-50 text-gray-400 material-symbols-outlined"
            style={{
              fontSize: '3rem',
            }}
          >
            add
          </span>
        </div>
      </div>
    </div>
  );
}

export function AccountsPage(props: AccountsPageProps) {
  const [showNoAccounts, setShowNoAccounts] = useState(true);
  const router = useRouter();
  useEffect(() => {}, []);

  return (
    <>
      {showNoAccounts && (
        <NoAccountsPage
          onAddClick={() => {
            router.push('/accounts/new');
          }}
        ></NoAccountsPage>
      )}
    </>
  );
}

export default AccountsPage;
