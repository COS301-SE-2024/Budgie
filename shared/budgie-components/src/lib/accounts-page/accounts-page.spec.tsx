import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { AccountsPage } from './accounts-page';
import { UserContext } from '@capstone-repo/shared/budgie-components';
import { useRouter } from 'next/navigation';
import { getDocs, collection } from 'firebase/firestore';
import { db } from '../../../../../apps/budgie-app/firebase/clientApp';

// Mocking dependencies
jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}));
jest.mock('firebase/firestore', () => ({
  collection: jest.fn(),
  query: jest.fn(),
  getDocs: jest.fn(),
}));
jest.mock('../../useThemes', () => ({
  useThemeSettings: jest.fn(),
}));

const mockRouterPush = jest.fn();

describe('AccountsPage', () => {
  const mockUser = { uid: 'user123' };

  beforeEach(() => {
    (useRouter as jest.Mock).mockReturnValue({ push: mockRouterPush });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

});
