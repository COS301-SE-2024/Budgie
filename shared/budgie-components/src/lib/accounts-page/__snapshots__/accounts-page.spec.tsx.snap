// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountsPage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mainPage"
  >
    <div
      class="flex items-center justify-center h-full w-full"
    >
      <div
        class="flex flex-col items-center justify-center bg-BudgieGray hover:bg-black hover:bg-opacity-5 h-80 w-80 border-dashed border-2 border-gray-400 border-opacity-40 hover:border-opacity-100 cursor-pointer  rounded-3xl"
      >
        <span
          class="text-2xl text-gray-400 text-opacity-60"
        >
          No Accounts Added Yet
        </span>
        <span
          class="mt-5 text-xl text-gray-400 text-opacity-60"
        >
          Add an Account
        </span>
        <span
          class="mt-2 text-opacity-50 text-gray-400 material-symbols-outlined"
          style="font-size: 3rem;"
        >
          add
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
